SeamCarver.js

```diff
@@ -203,17 +220,105 @@ class SeamCarver {
     }
 
     /**
+     * Removes vertical seam.
+     * Recalculates pixels depending on removed pixel.
+     *
+     */
+    removeVerticalSeam(vseam) {
+        this.imageData = this.context.createImageData(this.width - 1, this.height);
+        for (var row = this.height - 1; row >= 0; row--) {
+            var deletedCol = vseam[row];
+
+            // copy across pixels before seam col
+            for (var col = 0; col < deletedCol; col ++) {
+                var oldPos = this.pixelToIndex(col, row);
+                var pos = oldPos - (row * 4)
+                for (var i = 0; i < 4; i ++) {
+                    this.imageData.data[pos + i] = this.picture[oldPos + i];
+                }
+            }
+
+            // Start at deleted col
+            // Can ignore last column as we will delete it
+            for (var col = deletedCol; col < this.width - 1; col ++) {
+
+                // copy across pixels after seam col
+                var pos = this.pixelToIndex(col, row) - (row * 4);
+                var pos_right = this.pixelToIndex(col + 1, row);
+                for (var i = 0; i < 4; i ++) {
+                    this.imageData.data[pos + i] = this.picture[pos_right + i];
+                }
+
+                // copy across energy_matrix
+                var val_right = this.energy_matrix[col + 1][row];
+                val_right.minx--;
+                this.energy_matrix[col][row] = val_right;
+            }
+        }
+
+        this.energy_matrix.splice(this.width - 1, 1)
+        this.picture = this.imageData.data;
+        this.width--;
+
+        // now update energy matrix
+        // don't need to recalculate last row
+        for (var row = this.height - 2; row >= 0; row--) {
+            var deletedCol = vseam[row];
+
+            for (var i = -3; i < 4; i ++) {
```

Comment:
I think that, after removing the seam, you actually have to recalculate the whole matrix! Or else  vminsum might be wrong. A change of the energy in one pixel propagate upwards to all three possible parents, that means that a change on the last row might (theoretically) impact the minimum sum of any pixel on the top row - not only those adjacent to the one that was removed ...